includes/class-wc-stripe-intent-controller.php

```diff
@@ -304,14 +304,21 @@ public function create_payment_intent( $order_id = null ) {
 			$amount = $order->get_total();
 		}
 
-		$gateway = new WC_Stripe_UPE_Payment_Gateway();
+		$gateways = WC()->payment_gateways()->payment_gateways();
+		/**
+		 * Our gateway object.
+		 *
+		 * @var $gateway WC_Stripe_UPE_Payment_Gateway
+		 */
+		$gateway                 = $gateways[ WC_Stripe_UPE_Payment_Gateway::ID ];
+		$enabled_payment_methods = $gateway->get_upe_enabled_at_checkout_payment_method_ids();
```

Comment:
We already discussed that we need to move this methods (and those from the order controller) to the UPE Gateway, but it might be worth mentioning that both the existent Stripe Gateway and the new UPE Stripe Gateway have the same ID (`stripe`)... just to be safe we can add something like this:
```
if ( ! is_a( $gateway, 'WC_Stripe_UPE_Payment_Gateway' ) ) {
    throw new Exception( __( "We're not able to process this payment.", 'woocommerce-gateway-stripe' ) );
}
```